// Copyright (c) YugaByte, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License"); you may not
// use this file except in compliance with the License.  You may obtain a copy
// of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
// License for the specific language governing permissions and limitations under
// the License.
//

syntax = "proto3";

package yb.stateful_service;

option java_package = "org.yb.stateful_service";

import "yb/common/wire_protocol.proto";

service PgCronLeaderService {
  rpc PgCronSetLastMinute(PgCronSetLastMinuteRequestPB)
      returns (PgCronSetLastMinuteResponsePB);
  rpc PgCronGetLastMinute(PgCronGetLastMinuteRequestPB)
      returns (PgCronGetLastMinuteResponsePB);
}

message PgCronSetLastMinuteRequestPB {
  int64 last_minute = 1;
}

message PgCronSetLastMinuteResponsePB {
  AppStatusPB error = 1;
}

message PgCronGetLastMinuteRequestPB {
}

message PgCronGetLastMinuteResponsePB {
  int64 last_minute = 1;
  AppStatusPB error = 2;
}
